euphonic/spectra.py

```diff
@@ -669,10 +674,387 @@ def broaden(self: T, x_width,
         return new_spectrum
 
 
-LineData = Sequence[Dict[str, Union[str, int]]]
+OneLineData = Dict[str, Union[str, int]]
+LineData = Sequence[OneLineData]
+Metadata = Dict[str, Union[str, int, LineData]]
```

Comment:
Presumably not possible to build `TypedDict`s

Reply:
No, TypedDict is for when you know what the keys are. We have no opinion about which keys should have int values and which should have str values, so `Dict[str, str | int]` is exactly the right annotation.